### ISSUE TITLE
Send multiple measurements

### ISSUE THREAD

**Issue by hhj1995** [sitewhere/sitewhere#825]
How can sitewhere send multiple measured values at the same time through the mqtt client. The example given on the official website can only send one test value, not multiple

**Comment by derekadams** [comment]
Sending multiple measurements in one request is only supported in the REST APIs and not via MQTT at this time. See [this API call](https://sitewhere.io/docs/2.1.0/api2/#operation/addDeviceEventBatchUsingPOST) for information about sending multiple events.

**Comment by hhj1995** [comment]
Is the interface for sending multiple measurement values of version 2.0.0 encapsulated in the officially provided rest-api library


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Derek Adams"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月12日(星期五) 晚上9:58
收件人:&nbsp;"sitewhere/sitewhere"<sitewhere@noreply.github.com&gt;;
抄送:&nbsp;"霓裳挽歌倾城醉"<2638291929@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [sitewhere/sitewhere] Send multiple measurements (#825)





 
Sending multiple measurements in one request is only supported in the REST APIs and not via MQTT at this time. See this API call for information about sending multiple events.

**Comment by derekadams** [comment]
Yes, it's the `addMultipleEventsForDevice` method [here](https://github.com/sitewhere/sitewhere-java-api/blob/master/sitewhere-java-client/src/main/java/com/sitewhere/rest/client/SiteWhereClient.java#L1997). Note you can send any number of locations, alerts, and measurements in the payload.

**Comment by hhj1995** [comment]
In version 2.0.0, the device allocation information has been found in influcDB, but the inbound microservice still prompts that the device is not allocated, so the mqtt client cannot transmit messages into it.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Derek Adams"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月12日(星期五) 晚上10:20
收件人:&nbsp;"sitewhere/sitewhere"<sitewhere@noreply.github.com&gt;;
抄送:&nbsp;"霓裳挽歌倾城醉"<2638291929@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [sitewhere/sitewhere] Send multiple measurements (#825)





 
Yes, it's the addMultipleEventsForDevice method here. Note you can send any number of locations, alerts, and measurements in the payload.

**Comment by hhj1995** [comment]
Sorry to disturb you, I found that the official interface for transmitting multiple measured values is wrong. How to debug this situation when compiling the source code in idea?






------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Derek Adams"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月12日(星期五) 晚上10:20
收件人:&nbsp;"sitewhere/sitewhere"<sitewhere@noreply.github.com&gt;;
抄送:&nbsp;"霓裳挽歌倾城醉"<2638291929@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [sitewhere/sitewhere] Send multiple measurements (#825)





 
Yes, it's the addMultipleEventsForDevice method here. Note you can send any number of locations, alerts, and measurements in the payload.
